### PR TITLE
New script: Part-DB

### PR DESCRIPTION
**Pull request by bvdberg01**
> [!NOTE]
> We are meticulous when it comes to merging code into the main branch, so please understand that we may reject pull requests that do not meet the project's standards. It's never personal. Also, game-related scripts have a lower chance of being merged.

## Description
Adding script for [Part-DB](https://github.com/Part-DB/Part-DB-server) as requested in discussion https://github.com/community-scripts/ProxmoxVE/discussions/464.

I tested installation on my Proxmox installation. The update script is also tested by installing an older version from Part-DB on install (changing line 67 in install/part-db-install.sh to RELEASE="1.13.0") and then running the update script.

## Type of change
Please check the relevant option(s):

- [ ] Bug fix (non-breaking change that resolves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [x] New script (a fully functional and thoroughly tested script or set of scripts.)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [X] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [X] Testing performed (I have tested my changes, ensuring everything works as expected)
- [X] Documentation updated (I have updated any relevant documentation)

## Additional Information
**Installed LXC:**
lxc

**Installed old version incl. added testdata:**
before update1
before update

**Installing update:**
update

**After update:**
After update1
after update
Screenshot 2024-11-29 221717

## Related Pull Requests / Discussions
https://github.com/community-scripts/ProxmoxVE/discussions/464

### CODE REVIEW

**Review by newzealandpaul — DISMISSED** [review]

File: install/part-db-install.sh [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    </Directory>
+
+    ErrorLog /var/log/apache2/partdb_error.log
+    CustomLog /var/log/apache2/partdb_access.log combined
+</VirtualHost>
+EOF
+$STD a2ensite partdb
+$STD a2enmod rewrite
+rm /etc/apache2/sites-enabled/000-default.conf
+service apache2 restart
```
Comment (MickLesk) [inline install/part-db-install.sh:107]: $STD systemctl reload apache2
msg_ok "Created Service"

File: install/part-db-install.sh [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+echo "${RELEASE}" >/opt/${APPLICATION}_version.txt
```
Comment (MickLesk) [inline install/part-db-install.sh:108]: move above msg_ok "Installed Part-DB"

File: ct/part-db.sh [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+  msg_info "Updating $APP to v${RELEASE}"
+  mv /var/www/partdb/ /opt/partdb-backup
+  cd /opt
+  wget -q "https://github.com/Part-DB/Part-DB-server/archive/refs/tags/v${RELEASE}.zip"
+  unzip -q "v${RELEASE}.zip"
+  mv /opt/Part-DB-server-${RELEASE}/ /var/www/partdb
+  chown -R www-data:www-data /var/www/partdb
+
+  cd /var/www/partdb/
```
[the remark this answers is not in the document]
Comment (bvdberg01) [reply ct/part-db.sh:76]: Now partsdb lives in opt folder

File: ct/part-db.sh [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  cp -r "/opt/partdb-backup/.env.local" /var/www/partdb/
+  cp -r "/opt/partdb-backup/public/media" /var/www/partdb/public/
+  cp -r "/opt/partdb-backup/config/banner.md" /var/www/partdb/config/
+  
+  sudo -u www-data composer install --no-dev -o &>/dev/null
```
[the remark this answers is not in the document]
Comment (bvdberg01) [reply ct/part-db.sh:81]: Yes, but then  export COMPOSER_ALLOW_SUPERUSER=1 is needed

File: ct/part-db.sh [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  MTU=""
+  SD=""
+  NS=""
+  MAC=""
+  VLAN=""
+  SSH="no"
+  VERB="no"
+  echo_default
+}
+
```
Comment (MickLesk) [inline ct/part-db.sh:54]: can you migrate to new header? Example:

```bash
#!/usr/bin/env bash
source <(curl -s https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/misc/build.func)
# Copyright (c) 2021-2024 tteck
# Author: MickLesk (Canbiz) & vhsdream
# License: MIT | https://github.com/community-scripts/ProxmoxVE/raw/main/LICENSE
# Source: https://hoarder.app/

# App Default Values
APP="Hoarder"
var_tags="bookmark"
var_cpu="2"
var_ram="4096"
var_disk="8"
var_os="debian"
var_version="12"
var_unprivileged="1"

# App Output & Base Settings
header_info "$APP"
base_settings

# Core
variables
color
catch_errors
```

File: ct/part-db.sh [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+exit
+}
+
+start
+build_container
+description
+
+msg_ok "Completed Successfully!\n"
+echo -e "${APP} should be reachable by going to the following URL.
+         ${BL}http://${IP}${CL} \n"
```
Comment (MickLesk) [inline ct/part-db.sh:109]: can you migrate to new footer? Example:
```bash
msg_ok "Completed Successfully!\n"
echo -e "${CREATING}${GN}${APP} setup has been successfully initialized!${CL}"
echo -e "${INFO}${YW} Access it using the following URL:${CL}"
echo -e "${TAB}${GATEWAY}${BGN}http://${IP}:3000${CL}"
```

Comment (bvdberg01) [reply ct/part-db.sh]: Fixed

File: ct/part-db.sh (hunk 5 above)
Comment (bvdberg01) [reply ct/part-db.sh]: Fixed, but this example contains wrong copyright info

**Review by MickLesk — APPROVED** [review]
LGTM

**Review by michelroegl-brunner — APPROVED** [review]

**Pull request merged**